keps/sig-architecture/4330-compatibility-versions/README.md

```diff
@@ -562,6 +565,18 @@ func ClientFunction() {
 
 ```
 
+#### Non-Emulatable Features
+Generally, if a feature is enabled by default either in Beta or GA, it should be fairly stable already.
+In very rare cases for some default-off Beta features, a feature could be non-emulatable if the feature implementation history could not be preserved in the code base with reasonable maintenance cost.
+
+For these cases, we should try to emulate these changing features at best effort, and make sure the clients are aware of the risks of enabling default-off Beta features with emulation version.
+
+##### Alternatives
+An alternative would be to keep a exception list of `NonEmulatableFeatures`, and make the server fail to start if any of the `NonEmulatableFeatures` are enabled in compatibility mode. This approach has the benefits of predictable outcomes. The downsides are:
+1. whether a feature should be added to the list is subjective. It is hard to define a threshold to decide if the difficulty to preserve the implementation history is above this threshold, we could add the feature to the list.
+1. this could break automated upgrade processes using the emulation version, depending what features a cluster opts in.
+1. there needs proper cleaning up rules to ensure the list would not just keep growing.
```

Comment:
FYI Jordan and David did something similar for avoiding beta apis to be enabled by default https://github.com/kubernetes/kubernetes/commit/af99d192cf95cd29e455d186878db409b8cbdf2c